src/models/room.ts

```diff
@@ -914,37 +914,79 @@ export class Room extends ReadReceipt<RoomEmittedEvents, RoomEventHandlerMap> {
         return this.myUserId;
     }
 
+    /**
+     * Gets the "functional members" in this room.
+     *
+     * Returns the list of userIDs from the `io.element.functional_members` event. Does not consider the
+     * current membership states of those users.
+     *
+     * @see https://github.com/element-hq/element-meta/blob/develop/spec/functional_members.md.
+     */
+    private getFunctionalMembers(): string[] {
+        const mFunctionalMembers = this.currentState.getStateEvents(UNSTABLE_ELEMENT_FUNCTIONAL_USERS.name, "");
+        if (Array.isArray(mFunctionalMembers?.getContent().service_members)) {
+            return mFunctionalMembers!.getContent().service_members;
+        }
+        return [];
+    }
+
     public getAvatarFallbackMember(): RoomMember | undefined {
-        const memberCount = this.getInvitedAndJoinedMemberCount();
-        if (memberCount > 2) {
+        const functionalMembers = this.getFunctionalMembers();
+        // Optimizing for performance, we iterate over all members outermost, as is it the longest of the 3 lists we are comparing.
+        const nonFunctionalMemberCount = this.getMembers()!.reduce((count, m) => {
+            if (
+                m.membership &&
+                // We care most about big rooms where also possibly lots of users may have left,
+                // so we first do the O(2) membership check, as it is definitely cheap and might remove candidates before the next step.
+                ["join", "invite"].includes(m.membership) &&
+                // Then we do the O(functionalMembers.length) functionalMember check, which is probably usually still cheap.
+                !functionalMembers.includes(m.userId)
+            ) {
+                return count + 1;
+            }
+            return count;
```

Comment:
This seems a confusing way to write this. And tbh I wouldn't bother documenting the performance considerations in this much detail.

```suggestion
            if (m.membership !== 'join' && m.membership !== 'invite') return count;
            if (functionalMembers.includes(m.userId)) return count;
            return count+1;
```